### PR TITLE
Feature/dynamic rig loading

### PR DESCRIPTION
**Pull request by SimonTheSourcerer**
### Description
Allows to add an interaction rig on runtime, works with Unity XR

**Test Configuration**:
git clone --single-branch --branch develop git@github.com:Innoactive/Basic-Conditions-And-Behaviors.git Innoactive/Creator/Components/Basic-ConditionsBehaviors
git clone --single-branch --branch feature/dynamic-rig-loading git@github.com:Innoactive/Basic-Interaction-Component.git Innoactive/Creator/Components/Basic-Interaction
git clone --single-branch --branch develop git@github.com:Innoactive/Basic-UI-Component.git Innoactive/Creator/Components/Basic-UI-Components
git clone --single-branch --branch develop git@github.com:Innoactive/TextToSpeech-Component.git Innoactive/Creator/Components/TextToSpeech
git clone --single-branch --branch feature/dynamically-loaded-xr-rigs git@github.com:Innoactive/XR-Interaction-Component.git Innoactive/Creator/Components/XR-Interaction-Component
git clone --single-branch --branch develop git@github.com:Innoactive/Creator.git Innoactive/Creator/Core
git clone --single-branch --branch develop git@github.com:Innoactive/IA-Training-Template.git Innoactive/Creator/IA-Training-Template

### CODE REVIEW

File: Editor/RigSetup/InteractionFrameworkSceneSetup.cs [hunk 1]
```diff
@@ -1,6 +1,4 @@
-﻿using UnityEngine;
-
-namespace Innoactive.CreatorEditor.BasicInteraction
+﻿namespace Innoactive.CreatorEditor.BasicInteraction.RigSetup
```
Comment (Gusinuhe) [inline Editor/RigSetup/InteractionFrameworkSceneSetup.cs:1]: Screenshot 2021-01-26 at 13 13 43

**Review by Gusinuhe — CHANGES_REQUESTED** [review]

File: Editor/RigSetup/InteractionRigSetupDrawer.cs [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+                }
+                
+                bool canBeUsed = foundProvider == null || foundProvider[index].CanBeUsed();
+
+                Rect labelRect = new Rect(rect.x, rect.y, rect.width - 2 * lineHeight - 4, lineHeight);
+                GUI.enabled = canBeUsed;
+                EditorGUI.LabelField(labelRect, rigSetup.PossibleInteractionRigs[index].Name);
+                GUI.enabled = true;
+                
+                if (!canBeUsed)
```
Comment (Gusinuhe) [inline Editor/RigSetup/InteractionRigSetupDrawer.cs:53]:
```suggestion
                if (canBeUsed == false)
```

**Comment by Gusinuhe** [comment]
Funny log messages are spotted!

image

**Review by Gusinuhe — DISMISSED** [review]

**Review by Gusinuhe — APPROVED** [review]

**Comment by Luphoxa** [comment]
I cloned all the needed components in an existing project and I get the following errors. I had the same problem with another existing project on thursday and asked Gustavo for help, but he was not able to reproduce it. I will try it on Monday with a completely new project and probably ask for dev help on the topic, as this is a real pain ^^'

> Assets\Innoactive\Creator\Components\Basic-UI-Components\Runtime\CourseController\StandaloneMenuHandler.cs(72,25): error CS0234: The type or namespace name 'InputSystem' does not exist in the namespace 'UnityEngine' (are you missing an assembly reference?)

> Assets\Innoactive\Creator\Components\Basic-UI-Components\Runtime\CourseController\StandaloneMenuHandler.cs(72,108): error CS0234: The type or namespace name 'InputSystem' does not exist in the namespace 'UnityEngine' (are you missing an assembly reference?)

**Comment by Luphoxa** [comment]
I was still testing this, as said in the daily. If you are not sure if I am done testing or have tested this at all, feel free to write me in slack to ask or give me a headsup that I am already testing old news, please. Thanks.

**Pull request merged**